src/main/java/epam/gym/controller/TrainingController.java

```diff
@@ -1,41 +1,66 @@
 package epam.gym.controller;
 
 import epam.gym.dto.training.TrainingDto;
+import epam.gym.entity.ActionType;
+import epam.gym.entity.Training;
 import epam.gym.service.TrainingService;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
+import org.springframework.jms.core.JmsTemplate;
 import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
+import java.util.Optional;
+
+
 @RestController
 @RequestMapping("/trainings")
 public class TrainingController {
 
     private TrainingService trainingService;
 
+    @Value("${activemq.destination}")
+    private String destination;
+
+    private JmsTemplate jmsTemplate;
+
+
     TrainingController(TrainingService trainingService){
         this.trainingService = trainingService;
 
     }
 
     @PostMapping("/add")
     public ResponseEntity<String> addTraining(@RequestBody TrainingDto trainingDto) {
-        trainingService.create(trainingDto);
+        Training training = trainingService.create(trainingDto);
+        trainingService.convertAndSendToConsumer(training, destination, ActionType.ADD);
+
         return ResponseEntity.ok("Training added and notification sent.");
     }
 
     @DeleteMapping("/delete/{id}")
     public ResponseEntity<String> deleteTraining(@PathVariable Long id) {
-        boolean isDeleted = trainingService.delete(id);
-        if (isDeleted) {
-            return ResponseEntity.ok("Training deleted successfully.");
+
+        Optional<Training> trainingOptional = trainingService.findById(id);
```

Comment:
It is better to move all logic to service layer. Just call `trainingService.delete(id);`. If anything goes wrong in method delete, throw exception. Then in exception handler create response object like ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
                        .body("Training could not be deleted.");